DQM/src/DQM/VisiblesCutflow.cxx

```diff
@@ -0,0 +1,410 @@
+#include "DQM/VisiblesCutflow.h"
+
+// LDMX
+#include "DetDescr/EcalID.h"
+#include "DetDescr/HcalID.h"
+#include "DetDescr/SimSpecialID.h"
+#include "Ecal/Event/EcalHit.h"
+#include "Ecal/Event/EcalVetoResult.h"
+#include "Hcal/Event/HcalHit.h"
+#include "SimCore/Event/SimCalorimeterHit.h"
+#include "SimCore/Event/SimParticle.h"
+#include "SimCore/Event/SimTrackerHit.h"
+#include "Tracking/Event/Track.h"
+
+// C++
+#include <algorithm>
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <numeric>
+
+namespace dqm {
+
+void VisiblesCutflow::configure(framework::config::Parameters &parameters) {
+  rt_ = std::make_unique<ldmx::Ort::ONNXRuntime>(
+      parameters.get<std::string>("bdt_file"));
+  feature_list_name_ = parameters.get<std::string>("feature_list_name");
+  bdt_cut_val_ = parameters.get<double>("disc_cut");
+  ecal_bdt_cut_val_ = parameters.get<double>("ecal_disc_cut");
+  all_cuts_ = parameters.get<bool>("all_cuts");
+
+  beam_energy_mev_ = parameters.get<double>("beam_energy");
+
+  // collection names
+  hcal_rec_collection_ = parameters.get<std::string>("hcal_rec_coll_name");
+  hcal_rec_pass_name_ = parameters.get<std::string>("hcal_rec_pass_name");
+
+  ecal_rec_collection_ = parameters.get<std::string>("ecal_rec_coll_name");
+  ecal_rec_pass_name_ = parameters.get<std::string>("ecal_rec_pass_name");
+
+  recoil_from_tracking_ = parameters.get<bool>("recoil_from_tracking");
+  track_collection_ = parameters.get<std::string>("track_collection");
+  track_pass_name_ = parameters.get<std::string>("track_pass_name");
+
+  sp_collection_ = parameters.get<std::string>("sp_coll_name");
+  sp_pass_name_ = parameters.get<std::string>("sp_pass_name");
+  sim_particles_pass_name_ =
+      parameters.get<std::string>("sim_particles_pass_name");
+
+  ecal_veto_collection_ = parameters.get<std::string>("ecal_veto_coll_name");
+  ecal_veto_pass_ = parameters.get<std::string>("ecal_veto_pass_name");
+}
+
+bool VisiblesCutflow::inList(std::vector<int> parents, int track_id) {
+  bool in_list = false;
+  for (const int &i : parents) {
+    if (i == track_id) {
+      in_list = true;
+    }
+  }
+  return in_list;
+}
+
+void VisiblesCutflow::analyze(const framework::Event &event) {
+  std::vector<float> bdt_features;
+
+  double decay_z;
+
+  auto particle_map{event.getMap<int, ldmx::SimParticle>(
```

Comment:
Making this a `const` reference `&` prevents an extra copy on each event. If you want to edit the map/vector while filtering it, then making a copy makes sense. Just wanting to avoid it if its not necessary.

```suggestion
  const auto &particle_map{event.getMap<int, ldmx::SimParticle>(
```